src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -165,22 +165,29 @@ class DecisionTree(BaseModel):
         return np.asarray(self.predictor.predict_proba(X))
 
     def get_leaf_matrix(self, scale: bool = False) -> np.ndarray:
```

Comment:
To simplify the code a bit, would it be possible to merge the 'get_leaf_matrix' with the 'predict_leaf_matrix' and just run the get_leaf_matrix code if X is not provided? Or is there a good reason to keep them separate?